dds/idl/annotations.cpp

```diff
@@ -172,16 +208,15 @@ std::string KeyAnnotation::name() const
 
 bool KeyAnnotation::union_value(AST_Union* node) const
 {
-  return value_from_appl(node->disc_annotations().find(declaration()));
+  AST_Annotation_Appl* appl = node->disc_annotations().find(declaration());
+  if (!appl) { return absent_value; }
+  return value_from_appl(appl, default_value());
 }
 
 // @topic ====================================================================
 
 TopicAnnotation::TopicAnnotation()
 {
-  platforms_.insert("*");
-  platforms_.insert("DDS");
-  platforms_.insert("OpenDDS");
```

Comment:
Where are you putting these back in? I think I've missed it.

Reply:
Why are they necessary?

Reply:
They are part of the definition of `@topic`, though I admit they're not really all that important. Still I don't see a reason to remove them.

Reply:
The previous code only allowed topics with a platform in that set.  I think my interpretation of the spec is different.  The platform is a string or pattern that indicates that the topic is specifically for DDS or for any system (*).  This perhaps allows for optimization, or security, or something else that the spec writers dreamed about.

I'm okay putting it back it but I want to clarify the intended behavior.  Specifically for the error case.  If a specify a platform that is not in that list, should I see an error or a warning?

Reply:
> indicates that the topic is specifically for DDS or for any system (*)

By "topic" do you mean the topic it's talking about for the "name"? If so than maybe, but it's not giving an indication the two parameters are related.

Reply:
I agree that name and platform are not related unless a code generator takes actions based on the two, i.e., a generator that sees DDS for the platform always generates a sample reader and writer using the given topic name.

So, back to my original question:  Why are we whitelisting the platforms?  Why are we not printing an error for platforms that are not in the list?

Reply:
We don't always follow this rule, but the IDL spec says that annotations shouldn't really cause errors or warnings. In addition to that, I figured when this was set to something we don't recognize, it should be interpreted as an intentional action by the user. Although, like with unknown annotations, I wouldn't be against a configurable warning in that defaults to on in this case though.